extensions/common/vault/vault-hashicorp/src/test/java/org/eclipse/edc/vault/hashicorp/HashicorpVaultClientTest.java

```diff
@@ -20,217 +21,392 @@
 import static org.mockito.Mockito.when;
 
 class HashicorpVaultClientTest {
+
+    private static final String VAULT_TOKEN = UUID.randomUUID().toString();
     private static final String KEY = "key";
     private static final String CUSTOM_SECRET_PATH = "v1/test/secret";
     private static final String HEALTH_PATH = "sys/health";
     private static final Duration TIMEOUT = Duration.ofSeconds(30);
     private static final ObjectMapper OBJECT_MAPPER = new ObjectMapper();
-
-    private final EdcHttpClient edcClientMock = mock(EdcHttpClient.class);
+    private static final String VAULT_URL = "https://mock.url";
+    private static final HashicorpVaultConfig HASHICORP_VAULT_CLIENT_CONFIG = HashicorpVaultConfig.Builder.newInstance()
+            .url(VAULT_URL)
+            .secretPath(CUSTOM_SECRET_PATH)
+            .healthCheckPath(HEALTH_PATH)
+            .healthStandbyOk(false)
+            .token(VAULT_TOKEN)
+            .timeout(TIMEOUT)
+            .build();
+
+    private final EdcHttpClient httpClient = mock();
+    private final Monitor monitor = mock();
+    private final ScheduledExecutorService scheduledExecutorService = mock();
+    private final HashicorpVaultClient vaultClient = new HashicorpVaultClient(
+            HASHICORP_VAULT_CLIENT_CONFIG,
+            httpClient,
+            OBJECT_MAPPER,
+            scheduledExecutorService,
+            monitor);
 
     @BeforeEach
-    void setup() {
-
-    }
-
-    @Test
-    void getSecretValue() throws IOException {
-        // prepare
-        var vaultUrl = "https://mock.url";
-        var vaultToken = UUID.randomUUID().toString();
-        var config =
-                HashicorpVaultClientConfig.Builder.newInstance()
-                        .vaultUrl(vaultUrl)
-                        .vaultApiSecretPath(CUSTOM_SECRET_PATH)
-                        .vaultApiHealthPath(HEALTH_PATH)
-                        .isVaultApiHealthStandbyOk(false)
-                        .vaultToken(vaultToken)
-                        .timeout(TIMEOUT)
-                        .build();
-
-
-        var vaultClient = new HashicorpVaultClient(config, edcClientMock, OBJECT_MAPPER);
-        var response = mock(Response.class);
-        var body = mock(ResponseBody.class);
-        var payload = new GetEntryResponsePayload();
-
-        when(edcClientMock.execute(any(Request.class))).thenReturn(response);
-        when(response.code()).thenReturn(200);
-        when(response.body()).thenReturn(body);
-        when(body.string()).thenReturn(payload.toString());
-
-        // invoke
-        var result = vaultClient.getSecretValue(KEY);
-
-        // verify
-        assertNotNull(result);
-        verify(edcClientMock).execute(argThat(request -> request.method().equalsIgnoreCase("GET") &&
-                request.url().encodedPath().contains(CUSTOM_SECRET_PATH + "/data") &&
-                request.url().encodedPathSegments().contains(KEY)));
+    void beforeEach() {
+        reset(monitor);
```

Comment:
why do you need to reset the mock? it's not a static variable and gets initialized on every test

Reply:
if I don't reset it the verifications will fail